tests/lax_scipy_test.py

```diff
@@ -312,6 +314,116 @@ def apply_normalization(a):
       lsp_special_fn = lambda z: lsp_special.lpmn_values(l_max, l_max, z, is_normalized)
       self._CompileAndCheck(lsp_special_fn, args_maker)
 
+  def testSphHarmAccuracy(self):
+    m = jnp.array([-10, -2, 0, 1, 2])
+    n = jnp.array([20, 3, 1, 2, 6])
+    n_max = 20
+    theta = jnp.array([0.1, 0.1, 0.3, 0.5, 0.9])
+    phi = jnp.array([0.2, 0.2, 0.5, 1.0, 0.2])
+
+    expected = lsp_special.sph_harm(m, n, theta, phi, n_max)
+
+    actual = osp_special.sph_harm(m, n, theta, phi)
+
+    with self.subTest('Test real part.'):
+      self.assertAllClose(
+          jnp.real(actual), jnp.real(expected), rtol=1e-8, atol=9e-5)
+
+    with self.subTest('Test imaginary part.'):
+      self.assertAllClose(
+          jnp.imag(actual), jnp.imag(expected), rtol=1e-8, atol=9e-5)
+
+  def testSphHarmOrderZeroDegreeZero(self):
+    """Tests the spherical harmonics of order zero and degree zero."""
+    theta = jnp.array([0.3])
+    phi = jnp.array([2.3])
+    n_max = 0
+
+    expected = jnp.array([1.0 / jnp.sqrt(4.0 * np.pi)])
+    actual = jnp.real(
+        lsp_special.sph_harm(jnp.array([0]), jnp.array([0]), theta, phi, n_max))
+
+    self.assertAllClose(actual, expected, rtol=1.1e-7, atol=3e-8)
+
+  def testSphHarmOrderZeroDegreeOne(self):
+    """Tests the spherical harmonics of order one and degree zero."""
+    theta = jnp.array([2.0])
+    phi = jnp.array([3.1])
+    n_max = 1
+
+    expected = jnp.sqrt(3.0 / (4.0 * np.pi)) * jnp.cos(phi)
+    actual = jnp.real(
+        lsp_special.sph_harm(jnp.array([0]), jnp.array([1]), theta, phi, n_max))
+
+    self.assertAllClose(actual, expected, rtol=7e-8, atol=1.5e-8)
+
+  def testSphHarmOrderOneDegreeOne(self):
+    """Tests the spherical harmonics of order one and degree one."""
+    theta = jnp.array([2.0])
+    phi = jnp.array([2.5])
+    n_max = 1
+
+    expected = (-1.0 / 2.0 * jnp.sqrt(3.0 / (2.0 * np.pi)) *
+                jnp.sin(phi) * jnp.exp(1j * theta))
+    actual = lsp_special.sph_harm(
+        jnp.array([1]), jnp.array([1]), theta, phi, n_max)
+
+    with self.subTest('Test real part.'):
+      self.assertAllClose(
+          jnp.real(actual), jnp.real(expected), rtol=1e-8, atol=6e-8)
+
+    with self.subTest('Test imaginary part.'):
+      self.assertAllClose(
+          jnp.imag(actual), jnp.imag(expected), rtol=1e-8, atol=6e-8)
+
+  @parameterized.named_parameters(jtu.cases_from_list(
+      {'testcase_name': '_maxdegree={}_inputsize={}'.format(l_max, num_z),
+       'l_max': l_max, 'num_z': num_z}
+      for l_max, num_z in zip([1, 3, 8, 10], [2, 6, 7, 8])))
+  def testSphHarmForJitAndAgainstNumpy(self, l_max, num_z):
+    """Tests against JIT compatibility and Numpy."""
+    key = jrandom.PRNGKey(0)
```

Comment:
For tests, you can use a pattern like
```
rng = jtu.rand_int(self.rng(), -l_max, l_max + 1)
m = rng(shape, dtype)
```
You should be able to see this pattern throughout JAX's test suite.

Reply:
Done.